### ISSUE TITLE
Failed to create virtual branch from branch

### ISSUE THREAD

**Issue by mxmilkiib** [gitbutlerapp/gitbutler#3309]
So for reasons I removed a repo directory and recloned it. I loaded into Git Butler again, then I tried to "Apply" my previous remote PR branch, but recieved the error message "Failed to create virtual branch from branch". I then used `gh pr checkout [num]` to get the PR branch local, and then tried to Apply it in the UI but to the same result. I then did "`touch test` so there was some difference in the workspace, so as to give the columns, then tried to Apply, but to the same result.

Am I missing a step or such here? Or could Git Butler do something differently to make this work? Thanks!

**Comment by Byron** [comment]
Judging from my own experience, the way commits are currently displayed in a remote workflow can be confusing and in #3121 there is ongoing work to improve that display. With such overhaul, I would expect this behaviour to improve and become more intuitive.

In the meantime, when that happens again, maybe you could share [the errors from the application logs](https://docs.gitbutler.com/development/debugging#logs). Thank you.

**Comment by koppor** [comment]
This issue is the first google hit for `gitbutler gh pr checkout`.

When doing `gh pr checkout [num]`, gitbutler now says:

> Due to GitButler managing multiple virtual branches, you cannot switch back and forth between git branches and virtual branches easily. 

The linked documentation (https://docs.gitbutler.com/features/virtual-branches/integration-branch) does not have any hint on how to work on pull requests of other contributors.

I would love to do `gh pr checkout [num]` and then working with git butler on that base. Maybe, this refs https://github.com/gitbutlerapp/gitbutler/issues/4578?

**Comment by Byron** [comment]
In theory, once `gh pr checkout <id>` was run, there should be a new local branch tracking the branch on the remote repository from which the PR was created. That local branch should be suitable to bring into the workspace for edits. What's missing from there is the ability to actually interact with the underlying PR, or work with the underlying remote.

> I would love to do `gh pr checkout [num]` and then working with git butler on that base. Maybe, this refs #4578?

Maybe stacked virtual branches can also be used for that, but key here would be that the remote configuration of Git itself is respected.

**Comment by koppor** [comment]
> In theory, once `gh pr checkout {num}` was run, there should be a new local branch tracking the branch on the remote repository from which the PR was created.

Yeah! Currently, with 0.12.16, I get "Looks like you've switched away from `gitbutler/integration`". (I think, this is understandable, because the `gh` command leads to a checkout of a branch.)

**Comment by Caleb-T-Owens** [comment]
As an update to this issue, there are currently two known failure modes when applying existing branches:
- Applying branches that have resolved conflict markers
  - We can resolve this by either:
    - Cherry picking merge commits assuming the first parent (like `git cherry-pick -p 0 xxxx`) (this would cherry pick the right resolution).
    - Write out the re-merge as conflicted
    - Take inspiration from the re-merging approach that Elijah Newren demonstrated with `git replay` where you can try use the old merge result compared against the resolution as the new resolutions on top.
- Applying conflicted branches.
  - We provide a message to the user when this happens, but we should preemptively calculate the success and display that to the user so they can choose a desired resolution.

**Comment by Stephanie-Ziegler-Nanotec** [comment]
I've run into the same issue. After deleting all git hooks, the .git/gitbuttler directory and gitbuttler/workspace like branches, it worked fine again.

One small issue though? There aren't any log files? I don't know why it's not logging anything

**Comment by Byron** [comment]
That's a bit strange, but you should be able to conveniently open the logs folder like this:

<img width="579" height="479" alt="Image" src="https://github.com/user-attachments/assets/8b2739f0-3f06-45ba-8497-82187f6f9636" />

**Comment by Stephanie-Ziegler-Nanotec** [comment]
Ok, so they were under ~/.local/share/com.gitbutler.app.dev/logs/ (maybe update the kb?)

But... there's too much proprietary information in there... like unrelease products... I can't really upload them.\
And there's way too much to clean up...

**Comment by krlvi** [comment]
Hey! The branch-apply path behind this report has since been replaced: #13491 made the new apply implementation the default and removed the rebase that caused this class of failures. That shipped in `0.19.10`.

The `gh pr checkout` / regular-branch transition and stale workspace metadata cases were then fixed in #14517 and #14582, shipped in `0.21.0`.

I'm closing this as fixed. Please open a fresh issue with logs if applying a branch still fails on `0.21.0` or newer.
